src/components/@shared/FormFields/Provider/index.tsx

```diff
@@ -1,41 +1,47 @@
 import React, { ReactElement, useState } from 'react'
-import { ErrorMessage, useField } from 'formik'
+import { useField, useFormikContext } from 'formik'
 import UrlInput from '../URLInput'
 import { InputProps } from '@shared/FormInput'
 import FileInfo from '../FilesInput/Info'
 import styles from './index.module.css'
 import Button from '@shared/atoms/Button'
 import { initialValues } from 'src/components/Publish/_constants'
-import { ProviderInstance } from '@oceanprotocol/lib'
+import { LoggerInstance, ProviderInstance } from '@oceanprotocol/lib'
+import { FormPublishData } from 'src/components/Publish/_types'
 
 export default function CustomProvider(props: InputProps): ReactElement {
   const [field, meta, helpers] = useField(props.name)
   const [isLoading, setIsLoading] = useState(false)
+  const { setFieldError } = useFormikContext<FormPublishData>()
 
-  async function validateProvider(url: string) {
-    setIsLoading(true)
+  async function handleValidation(e: React.SyntheticEvent) {
+    e.preventDefault()
 
     try {
-      const isValid = await ProviderInstance.isValidProvider(url)
-      helpers.setValue({ url, valid: isValid })
-      helpers.setError(undefined)
+      setIsLoading(true)
+      const isValid = await ProviderInstance.isValidProvider(field.value.url)
+
+      // error if something's not right from response
+      // TODO: Figure out way to detect failed response, and throw that before
+      // this check.
+      // No way to detect a failed response with ProviderInstance.isValidProvider,
+      // making this error show up for multiple cases it shouldn't, namely network
```

Comment:
This is not accurate and overcomplicating things. If a provider is down then it is not a valid provider.  This is not a debug software for your web services/infrastructure.

Reply:
It's the very basic of web requests to know why they fail and we destroyed that here. If user's network is down, saying the entered provider is not valid is simply wrong but yes, considering custom provider is advanced use case anyway we don't need to bother